e2e_test/e2e_test.go

```diff
@@ -189,3 +191,89 @@ func TestStartStopValidators(t *testing.T) {
 	require.NoError(t, err)
 
 }
+
+type rpcCustomTransaction struct {
+	BlockNumber *hexutil.Big `json:"blockNumber"`
+	GasPrice    *hexutil.Big `json:"gasPrice"`
+}
+
+func TestRPCDynamicTxGasPriceWithState(t *testing.T) {
```

Comment:
I think we can drop this test because the the other test is practically already performing this test (except for checking the gasprice with respect to the tip and cap)

Reply:
I will keep it, but change the logic to assume it is an archive node, asking both, at the beginning and after N blocks. That way we will cover that case too